newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -1259,6 +1272,36 @@ export default class AuthenticatedUserProvider extends React.Component<
     });
   };
 
+  _onUpdateGithubStar = async (
+    githubUsername: string,
+    preferences: PreferencesValues
+  ) => {
+    const { authentication } = this.props;
+
+    await this._doEdit(
+      {
+        githubUsername,
+      },
+      preferences
+    );
+
+    this.setState({
+      editInProgress: true,
+    });
+    try {
+      const response = authentication.updateGitHubStar(
```

Comment:
You should await this, otherwise the `finally` clause is called instantly and `editInProgress` is never set to `true`